ui/lib/routes/transaction_validation_route/transaction_validation_form.dart

```diff
@@ -0,0 +1,221 @@
+    fromController = TextEditingController();
+    toController = TextEditingController();
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    amountController.dispose();
+    currencyController.dispose();
+    countryController.dispose();
+    fromController.dispose();
+    toController.dispose();
+    super.dispose();
+  }
+
+  void onCurrencySelect(BuildContext context) {
+    showCurrencyPicker(
+      context: context,
+      showFlag: true,
+      showCurrencyName: true,
+      showCurrencyCode: true,
+      onSelect: (Currency currency) {
+        setState(() => currencyController.text = currency.code);
+      },
+    );
+  }
+
+  void onContrySelect(BuildContext context) {
+    showCountryPicker(
+      context: context,
+      onSelect: (Country country) {
+        setState(() {
+          countryController.text = country.countryCode;
+          countryName = country.name;
+        });
+      },
+    );
+  }
+
+  Future<void> onValidateTransaction(BuildContext context) async {
+    if (amountController.text == '' ||
+        currencyController.text == '' ||
+        countryController.text == '' ||
+        fromController.text == '' ||
+        toController.text == '') {
+      showDialog(
+          context: context,
+          builder: (context) =>
+              const InfoDialog(text: 'No empty fields are allowed'));
+      return;
+    }
+
+    if (int.tryParse(amountController.text) == null ||
+        int.tryParse(toController.text) == null ||
+        int.tryParse(fromController.text) == null) {
+      showDialog(
+          context: context,
+          builder: (context) => const InfoDialog(
+              text:
+                  'Amount, to Id and from Id fields should have numerical values'));
+      return;
+    }
+
+    setState(() => isLoading = true);
+    try {
+      final response = await locator
+          .get<FraudDetectService>()
+          .transactionValidationServiceClient
+          .validateTransaction(
+            Transaction(
+                amount: Amount(
+                    amount: int.tryParse(amountController.text),
+                    currency: currencyController.text),
+                toAccount: toController.text,
+                fromAccount: fromController.text,
+                country: countryController.text),
+          );
+      if (context.mounted) {
+        if (response.status == ValidationStatus.SUCCESS) {
+          showDialog(
+              context: context,
+              builder: (context) =>
+                  const InfoDialog(text: 'Successful validation'));
+        } else {
+          showDialog(
+              context: context,
+              builder: (context) =>
+                  const InfoDialog(text: 'Validation failed'));
+        }
+      }
+    } catch (e) {
+      showDialog(
+          context: context,
+          builder: (context) => const InfoDialog(
+              text: 'An error happened during transaction validation'));
+    }
```

Comment:
We can reduce code duplicaton by creating new function called `showInfoDialog` which will accept text.